test/e2e/components_upgrade_test.go

```diff
@@ -0,0 +1,70 @@
+// +build e2e
```

Comment:
I think these functions could be generally useful to the e2e test suite, consider if we should put them in their own file independent of the `components_upgrade_test.go` file, perhaps in their own file in the Framework?